### PR TITLE
OQC<>Nvidia rest target integration

### PR DESCRIPTION
**Pull request by jfriel-oqc**
This first PR is as much a sanity check of my cpp!

This adds the OQC build target and allows compilation to target us. 

Once this is merged with our platform + tests to follow we can further upstream.

Whilst this is being reviewed I will work on building the test suite, it's kind of gnarly because we basically need a tiny mocked out version of the cloud. Not hard, but want to get it done correctly such that we're confident changes going forward won't break our integration

Can't see everyone on github yet to tag

### CODE REVIEW

File: unittests/backends/oqc/OQCTester.cpp [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+
+
+#include "CUDAQTestUtils.h"
+#include "common/FmtCore.h"
+#include "cudaq/algorithm.h"
+#include <fstream>
+#include <gtest/gtest.h>
+
+std::string mockPort = "62454";
+std::string backendStringTemplate =
```
Comment (oarnold-oqc) [inline unittests/backends/oqc/OQCTester.cpp:17]: @jfriel-oqc this hasn't been updated for OQC?

Comment (jfriel-oqc) [reply unittests/backends/oqc/OQCTester.cpp:17]: no, the full testing needs the little python server helper stuff. Working on that whilst the cpp stuff is reviewed

File: runtime/cudaq/platform/default/rest/helpers/oqc/OQCServerHelp.cpp [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+  return backendConfig.find(key) != backendConfig.end();
+}
+
+std::vector<std::string> OQCServerHelper::getJobID(int n){
+    RestHeaders headers = this -> getHeaders();
+    return client.get(backendConfig.at("job_path")+"/?" + std::to_string(n), "", headers );
+}
+
+std::string OQCServerHelper::makeConfig(int shots){
+    return "{\"$type\": \"<class 'scc.compiler.config.CompilerConfig'>\", \"$data\": {\"repeats\": "+std::to_string(shots)+", \"repetition_period\": null, \"results_format\": {\"$type\": \"<class 'scc.compiler.config.QuantumResultsFormat'>\", \"$data\": {\"format\": {\"$type\": \"<enum 'scc.compiler.config.InlineResultsProcessing'>\", \"$value\": 1}, \"transforms\": {\"$type\": \"<enum 'scc.compiler.config.ResultsFormatting'>\", \"$value\": 3}}}, \"metrics\": {\"$type\": \"<enum 'scc.compiler.config.MetricsType'>\", \"$value\": 6}, \"active_calibrations\": [], \"optimizations\": {\"$type\": \"<class 'scc.compiler.config.Tket'>\", \"$data\": {\"tket_optimizations\": {\"$type\": \"<enum 'scc.compiler.config.TketOptimizations'>\", \"$value\": 30}}}}}";
```
Comment (oarnold-oqc) [inline runtime/cudaq/platform/default/rest/helpers/oqc/OQCServerHelp.cpp:124]: Here and else where:

Define a `std::map<std::string, std::string>` fill with entries ` = { {key, value}, {key, value} }` then just `nlohmann::json_v3_11_1::json(map)`

I'll push some changes with similar in.

File: unittests/backends/oqc/OQCTester.cpp (hunk 1 above)
Comment (oarnold-oqc) [reply unittests/backends/oqc/OQCTester.cpp:17]: yeah sorry you did say in description.

**Pull request closed without merging**